pkg/auth/service.go

```diff
@@ -1902,13 +1903,44 @@ func NewAPIAuthService(apiEndpoint, token string, secretStore crypt.SecretStore,
 	} else {
 		cache = &DummyCache{}
 	}
-	res := &APIAuthService{
+	healthRes := &APIAuthService{
 		apiClient:   client,
 		secretStore: secretStore,
 		logger:      logger,
 		cache:       cache,
 	}
-	return res, nil
+
+	// Perform health check for API auth service
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = 5 * time.Second
+	bo.InitialInterval = 1 * time.Second
+	bo.MaxElapsedTime = 20 * time.Second
+	ctx := context.Background()
+	err = backoff.Retry(func() error {
+		healtResp, err := client.HealthCheck(ctx)
+		switch {
+		case err != nil:
+			return err
+		case healtResp.StatusCode == http.StatusNoContent:
+			return nil
+		default:
+			return fmt.Errorf("health check returned status %s: %w", healtResp.Status, ErrInvalidResponse)
+		}
+	}, bo)
+	if err != nil {
+		return nil, fmt.Errorf("health check failed: %w", err)
+	}
+
+	resp, err := client.GetVersionWithResponse(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get version failed: %w", err)
+	}
+	if resp.JSON200 == nil {
+		return nil, fmt.Errorf("get version returned status %s: %w", resp.Status(), ErrInvalidResponse)
+	}
+	logger.Info("auth API server version: ", resp.JSON200.Version)
```

Comment:
Suggest to move the code `GetVersion` method.

We can call the method during New, but I also thing it may not be a case for every usage. Like our database client - we require calling Ping after New to verify there is someone on the other side.